luigi/file.py

```diff
@@ -43,8 +49,55 @@ def move_to_final_destination(self):
     def generate_tmp_path(self, path):
         return path + '-luigi-tmp-%09d' % random.randrange(0, 1e10)
 
+class atomic_file_append(atomic_file):
+    """Use this to append to a temporary file and append to the final
+    destination when it is closed.
+    The destination file is locked to avoid concurrent writing to it.
+    """
+    def __init__(self, path, timeout=20, delay=.05):
+        self.timeout = timeout
+        self.delay = delay
+        self.is_locked = False
+        self.lockfile = "%s.lock" % path
+        super(atomic_file_append, self).__init__(path)
+
+    def move_to_final_destination(self):
+        try:
+            self.acquire()
+            with open(self.tmp_path) as in_file:
+                with open(self.path, 'a') as out_file:
+                    shutil.copyfileobj(in_file, out_file)
+        except Exception as exc:
+            self.release()
+            os.unlink(self.tmp_path)
+            raise exc
+        self.release()
+        os.unlink(self.tmp_path)
+
+    def acquire(self):
+        start_time = time.time()
+        while True:
+            try:
+                self.fd = os.open(
```

Comment:
Are you assuming a local file system for everything here? Will this work for hadoop?